src/file-actions.js

```diff
@@ -0,0 +1,30 @@
+import { FileAction, registerFileAction } from '@nextcloud/files'
+import { spawnDialog } from '@nextcloud/dialogs'
+import FileActionImport from './modules/modals/FileActionImport.vue'
+import tablesIcon from '@mdi/svg/svg/table-large.svg?raw'
+
+const validMimeTypes = [
+	'text/csv',
+	'text/html',
+	'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+	'application/vnd.ms-excel',
+]
+
+const fileAction = new FileAction({
+	id: 'import-to-tables',
+	displayName: () => t('tables', 'Import into Tables'),
+	iconSvgInline: () => tablesIcon,
+
+	enabled: (files) => {
+		const file = files[0]
+
+		return file.type === 'file' && validMimeTypes.includes(file.mime)
+	},
+
+	exec: (file) => {
+		spawnDialog(FileActionImport, { file })
+		return null
+	},
```

Comment:
We could possibly use [dynamic imports](https://webpack.js.org/guides/code-splitting/#dynamic-imports) to reduce the size of the entry point that is loaded and parsed on the files app. That way webpack can split the javascript chunks and this helps keeping the initial loading time sane:

Haven't tested but something like this should work (and removing the original import):
```suggestion
	async exec: (file) => {
		const { default: FileActionImport } = await import('./modules/modals/FileActionImport.vue')
		spawnDialog(FileActionImport, { file })
		return null
	},
```